src/LogicApp/LogicApp/Cmdlets/LogicApp/GetAzureLogicAppRunHistoryCommand.cs

```diff
@@ -55,8 +65,17 @@ public override void ExecuteCmdlet()
             base.ExecuteCmdlet();
             if (string.IsNullOrEmpty(this.RunName))
             {
-                var enumerator = LogicAppClient.GetWorkflowRuns(this.ResourceGroupName, this.Name).GetEnumerator();
-                this.WriteObject(enumerator.ToIEnumerable<WorkflowRun>(), true);
+                Page<WorkflowRun> page = LogicAppClient.GetWorkflowRuns(this.ResourceGroupName, this.Name);
+                this.WriteObject(page.GetEnumerator().ToIEnumerable<WorkflowRun>(), true);
+
+                if (FollowRelLink && !string.IsNullOrWhiteSpace(page.NextPageLink))
+                {
+                    for (int i = 0; i < MaximumFollowRelLink; i++)
+                    {
+                        page = LogicAppClient.GetWorkflowRuns(this.ResourceGroupName, this.Name, page.NextPageLink);
+                        this.WriteObject(page.GetEnumerator().ToIEnumerable<WorkflowRun>(), true);
```

Comment:
Best not to duplicate this block of code between first page and iteration. Consider refactoring the code to always enter and perform the first execution, while only iterating further when the next page link following is enabled.